doc/tutorials/namespaces.md

```diff
@@ -1,27 +1,93 @@
-# Namespaces #
+# Understanding Namespaces #
 
-## Definition ##
+## Structure of the key database ##
 
-In Elektra multiple keys may refer to the same part of the configuration.
-A _namespace_ is the part of the key that refers to the key's origin (like a unique path).
+The _key database_ of Elektra is _hierarchically structured_. This means that keys are organized similar to directories in a file system.
 
-Possible sources are: 
+Lets add some keys to the database. To add a key we can use this command:
 
-- **spec/something** for specification of other keys
-- **proc/something** for in-memory keys (e.g. command-line)
-- **dir/something** for dir keys in current working directory
-- **system/something** for system keys in /etc or /
-- **user/something** for user keys in home directory
-- **/something** for cascading keys (actually refers to one of the above)
+	kdb set <key> <value>
 
-Having namespaces enables both admins and users to set specific parts of the application's configuration.
+Now add the the key **/a** with the Value **Value 1** and the key **/b/c** with the Value **Value 2**:
+
+	kdb set /a 'Value 1'
+	kdb set /b/c 'Value 2'
+
+![alt text](img/elektra_hierarchy.svg "Hierarchical structure of key database")
```

Comment:
Please place the images to `doc/images`, then we have all of them together and can use them in different places.

@Namoshek Do you have an answer about the Markdown construct? I think at the moment we mostly use the syntax as shown above, but seems like we replaced it to html's `<img src=` for the main README.md?